File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    static String[] nums;
    public String solution(int[] numbers) {
        String answer = "";
        nums = new String[numbers.length];
        
        for(int i = 0; i < numbers.length; i++){
            nums[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(nums, new StringComparator());
        
        for(int i = 0; i < nums.length; i++){
            answer += nums[i];
        }
        if(answer.charAt(0) == '0')
            return "0";
        return answer;
    }
}

class StringComparator implements Comparator<String>{
        @Override
        public int compare(String a, String b) {
        return (b+a).compareTo(a+b);
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> temp;

bool cmp(string A, string B){
    return (A + B) > (B + A);
}

string solution(vector<int> numbers) {
    string answer = "";
    for(int i = 0; i < numbers.size(); i++){
        temp.push_back(to_string(numbers[i]));
    }
    
    sort(temp.begin(), temp.end(), cmp);
    
    for(int i = 0; i < temp.size(); i++){
        answer += temp[i];
    }
    
    if(answer[0] == '0')
        answer = "0";
    
    return answer;
}